tools/check_volume_levels.m

```diff
@@ -251,3 +279,30 @@ function plot_levels(meas, tc, lm)
 		y(:,j) = filter(b, a, x(:,j));
 	end
 end
+
+% This function is copy of
+% sof/tools/test/audio/test_utils/multitone.m
+function x = multitone( fs, f, amp, tlength )
+	n = round(fs*tlength);
+	t = (0:n-1)/fs;
+	nf = length(f);
+	if nf > 1
+		ph = rand(nf, 1)*2*pi;
+	else
+		ph = 0;
+	end
+
+	x = zeros(n, 1);
+	x(:,1) = amp(1)*sin(2*pi*f(1)*t);
+	for i=2:length(f)
+		x(:,1) = x(:,1) + amp(i)*sin(2*pi*f(i)*t+ph(i))';
```

Comment:
I think x(:,1) is no longer needed , Just x is sufficient

Reply:
Yep,

Reply:
It's now simplified to just `x` . I was apparently worried about vector orientations when doing this and did not clean up afterwards. The mixing of sine preserved  the vector and did not make it a 2D matrix (a pitfall with Matlab language with column and row vectors).